examples/nlp/training/trainer.py

```diff
@@ -0,0 +1,141 @@
+    TFBaseTrainerStep
+from transformers import TFDistilBertForSequenceClassification
+from zenml.utils.post_training.post_training_utils import \
+    get_feature_spec_from_schema
+
+
+class UrduTrainer(TFBaseTrainerStep):
+    def __init__(self,
+                 model_name: Text,
+                 serving_model_dir: Text = None,
+                 transform_output: Text = None,
+                 train_files: List[Text] = None,
+                 eval_files: List[Text] = None,
+                 batch_size: int = 64,
+                 epochs: int = 25,
+                 learning_rate: float = 1e-4,
+                 schema_file: Text = None,
+                 **kwargs
+                 ):
+
+        super(UrduTrainer, self).__init__(model_name=model_name,
+                                          serving_model_dir=serving_model_dir,
+                                          transform_output=transform_output,
+                                          train_files=train_files,
+                                          eval_files=eval_files,
+                                          batch_size=batch_size,
+                                          epochs=epochs,
+                                          learning_rate=learning_rate,
+                                          **kwargs)
+
+        # ML variables
+        self.batch_size = batch_size
+        self.epochs = epochs
+        self.learning_rate = learning_rate
+        self.model_name = model_name
+
+        if schema_file:
+            self.schema_path = os.path.dirname(schema_file)
+        else:
+            self.schema_path = None
+        self.schema_file = schema_file
+
+    def get_run_fn(self):
+        return self.run_fn
+
+    def run_fn(self):
+        feature_spec = get_feature_spec_from_schema(self.schema_path)
+        train_dataset = self.input_fn(self.train_files, feature_spec)
+        eval_dataset = self.input_fn(self.eval_files, feature_spec)
+
+        model = self.model_fn(train_dataset=train_dataset,
+                              eval_dataset=eval_dataset)
+
+        model.save_pretrained(self.serving_model_dir, saved_model=True)
+
+    def model_fn(self,
+                 train_dataset: tf.data.Dataset,
+                 eval_dataset: tf.data.Dataset):
+
+        id2label = {0: "FAKE_NEWS", 1: "REAL_NEWS"}
+
+        model = TFDistilBertForSequenceClassification.from_pretrained(
+            self.model_name, id2label=id2label)
+
+        optimizer = tf.keras.optimizers.Adam(learning_rate=self.learning_rate)
+
+        model.compile(optimizer=optimizer,
+                      loss=model.compute_loss,
+                      metrics=["accuracy"])
+
+        model.fit(train_dataset,
+                  epochs=self.epochs,
+                  callbacks=[
+                      tf.keras.callbacks.TensorBoard(
+                        log_dir=os.path.join(self.log_dir, 'train'))])
+
+        return model
+
+    def input_fn(self,
```

Comment:
If this is the same as in the Tensorflow Base Trainer, you can inherit from that class and therefore avoid duplicated code